### PR TITLE
[OD-606] Enhance harvester to import only specific organization/publisher datasets

### PR DESCRIPTION
**Pull request by jguo144**
[[OD-606] Enhance harvester to import only specific organization/publisher datasets feature](https://opengovinc.atlassian.net/browse/OD-606)

## Description
This PR enhances the data.json harvester to import datasets from specific organizations/publishers, define default values for dataset fields, and map fields from the data.json file to dataset fields.
These configuration options are set using a JSON object in the configuration form field in the harvester create/edit page.

Note: This PR also contains a bug fix to properly update existing resources, rather than deleting and creating new ones every harvest.

## Test Procedure
Install extensions
- Install the ckanext-harvest extension
- Install the ckanext-dcat extension
- Install this PR
Note: Remember to restart the supervisor service after any code changes to the harvest or dcat extension

1) Harvest datasets only from certain organizations
- Go to /harvest and click `Add Harvest Source`
- Set `URL` to https://data.ca.gov/data.json
- Set `Source type` to DCAT JSON Harvester
- Select an organization from the `Organization` dropdown menu
- Use the following configuration to include organizations, define default values and map fields
```
{
  "default_values": [
    {"license_id": "other-pd"}
  ],
  "map_fields": [
    {
      "source": "landingPage",
      "target": "url",
      "default": ""
    }
  ],
  "contact_point": {
    "name_field": "maintainer",
    "default_name": "DWR",
    "email_field": "maintainer_email",
    "default_email": "dwr@water.ca.gov"
  },
  "organizations_filter_include": ["California Department of Water Resources", "California Department of Water Resources (DWR)"]
}
```

2) Harvest datasets excluding datasets from organizations
- Repeat the steps for the previous harvest job, but use the following configuration to exclude organizations
```
{
  "organizations_filter_exclude": ["California Department of Fish and Wildlife", "Office of Statewide Health Planning & Development", "California Department of Public Health", "Department of Health Care Services", "California Department of Water Resources", "California Energy Commission", "Ocean Protection Council", "California State Water Resources Control Board"]
}
```

## Submitter Checklist
- [x] I have tested the changes locally

## Reviewer Checklist
- [x] I have run this code change locally
- [x] The description clearly states the impact of this code change.
- [x] There is enough information to verify the functionality

### CODE REVIEW

File: ckanext/dcat/harvesters/base.py [hunk 1]
```diff
@@ -422,6 +484,20 @@ def import_stage(self,harvest_object):
         if not package_dict.get('name'):
             package_dict['name'] = self._get_package_name(harvest_object, package_dict['title'])
 
+        # copy across resource ids from the existing dataset, otherwise they'll
+        # be recreated with new ids
```
Comment (ankitbrana) [inline ckanext/dcat/harvesters/base.py:488]: Looked over this block of code, but I'm having trouble understanding how this prevents the resources from being recreated with new ids? Could you elaborate further for my understanding

File: ckanext/dcat/harvesters/rdf.py [hunk 2]
```diff
@@ -66,27 +66,6 @@ def _get_guid(self, dataset_dict, source_url=None):
 
         return guid
 
-    def _get_existing_dataset(self, guid):
```
Comment (ankitbrana) [inline ckanext/dcat/harvesters/rdf.py:69]: I see that we moved this method to base.py, do we need to update the function call in rdf.py at line 285

**Review by ankitbrana — COMMENTED** [review]
Just a few questions, but overall looks good!

File: ckanext/dcat/harvesters/rdf.py (hunk 2 above)
Comment (jguo144) [reply ckanext/dcat/harvesters/rdf.py:69]: If you look on line 25 you can see that base is the parent of DCATRDFHarvester. I think we can leave the function call alone.

File: ckanext/dcat/harvesters/base.py (hunk 1 above)
Comment (jguo144) [reply ckanext/dcat/harvesters/base.py:488]: On line 499 the id of a resource is being set if the resource already exists.

**Review by ankitbrana — APPROVED** [review]
Looks good after the clarification! Tests ran smoothly locally.

**Pull request merged**